boot/assets.go

```diff
@@ -289,14 +365,161 @@ type TrustedAssetsUpdateObserver struct{}
 //
 // Implements gadget.ContentUpdateObserver.
 func (o *TrustedAssetsUpdateObserver) Observe(op gadget.ContentOperation, affectedStruct *gadget.LaidOutStructure, root, realSource, relativeTarget string) (bool, error) {
+	var whichBootloader bootloader.Bootloader
+	var whichAssets []string
+	var err error
+	var isRecovery bool
+
+	switch affectedStruct.Role {
+	case gadget.SystemBoot:
+		if o.bootBootloader == nil {
```

Comment:
we probably want different flags on o for this to avoid checking again if the bootloader has no trusted assets

Reply:
I'm wrong here, I was confused reading by the old name and vars of  findMaybeTrustedAssetsBootloader